autoload/ale/path.vim

```diff
@@ -17,6 +17,33 @@ function! ale#path#Simplify(path) abort
     return substitute(simplify(l:win_path), '^\\\+', '\', 'g') " no-custom-checks
 endfunction
 
+" Given a buffer and a file or directory name list, find the nearest file or directory by searching upwards
+" through the paths relative to the given buffer.
+function! ale#path#FindNearestMarker(buffer, markers) abort
+    let l:buffer_filename = fnamemodify(bufname(a:buffer), ':p')
+    let l:buffer_filename = fnameescape(l:buffer_filename)
+
+    for l:marker in a:markers
```

Comment:
Use a `for` loop with `ale#path#Upwards` instead. Have a look at the functions in the `python.vim` file for examples.